### DISCUSSION TITLE
Bulk Upsert with ID as primary key and Identity.

### DISCUSSION THREAD

**Discussion by RafaelCENG** [supabase/supabase#18334] [Questions]
Hello everyone I'm trying to do a bulk upsert. In the docs says
- Primary keys must be included in values to use upsert.
But my primary key is an identity that automatically assign a sequential unique number to the column.
How can i make it without taking the primary key? Now im hardcoding my id (as index) which is wrong
Here is my final Values array.
```
 [
                    {
                      id: i,
                      primary_name: pos.primary_name,
                      secondary_name: pos.secondary_name,
                    },
                    {
                      id: i + 1,    
                      primary_name: pos.primary_name,
                      secondary_name: "Not " + pos.secondary_name,
                    },
                    {....} etc
 ]
```
And here is my supabase call
```
  const saveCustomPositionsPromise = useCallback(
    (
      values: {
        id: number;
        primary_name: string;
        secondary_name: string;
      }[]
    ) => supabase.from("custom_positions").upsert(values).throwOnError(),
    []
  );
  ```

**Comment by GaryAustin1** [comment]
This has a discussion on it and some suggestions.
https://github.com/PostgREST/postgrest/issues/1118

**Reply by RafaelCENG** [reply]
First, thank you for the reply. Apparently from the discussion I didn't figure out something that could help me with my problem. Thanks again.
